pkg/apis/migration/v1alpha1/timed.go

```diff
@@ -0,0 +1,53 @@
+package v1alpha1
+
+import meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// Timed records started and completed timestamps.
+type Timed struct {
+	// Started timestamp.
+	Started *meta.Time `json:"started,omitempty"`
+	// Completed timestamp.
+	Completed *meta.Time `json:"completed,omitempty"`
+}
+
+// MarkReset mark as reset.
+func (r *Timed) MarkReset() {
+	r.Started = nil
+	r.Completed = nil
+}
+
+// MarkStarted mark as started.
+func (r *Timed) MarkStarted() {
+	if r.Started == nil {
+		r.Started = r.now()
+		r.Completed = nil
```

Comment:
pull this out of the if condition? Completed should be reset `nil` every time this function is called right?

Reply:
not really, MarkStarted() should only mark a Step started, it shouldn't worry about whether a step is completed or not. 
but when it does start a step, it should make sure the completed is not set by mistake

Reply:
aah I see. So you want to make MarkStarted idempotent, cool